sdk/storage/Azure.Storage.Files.DataLake/src/Models/ChangeAccessControlListResult.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+namespace Azure.Storage.Files.DataLake.Models
+{
+    /// <summary>
+    /// ChangeAccessControlListResult contains summary counts of operations that change Access Control Lists recursively.
+    /// See <see cref="DataLakePathClient.SetAccessControlListRecursive(System.Collections.Generic.IList{PathAccessControlItem}, int?, System.IProgress{ChangeAccessControlListPartialResult}, System.Threading.CancellationToken)"/>
+    /// <see cref="DataLakePathClient.SetAccessControlListRecursiveAsync(System.Collections.Generic.IList{PathAccessControlItem}, int?, System.IProgress{ChangeAccessControlListPartialResult}, System.Threading.CancellationToken)"/>
+    /// <see cref="DataLakePathClient.ModifyAccessControlListRecursive(System.Collections.Generic.IList{PathAccessControlItem}, int?, System.IProgress{ChangeAccessControlListPartialResult}, System.Threading.CancellationToken)"/>
+    /// <see cref="DataLakePathClient.ModifyAccessControlListRecursiveAsync(System.Collections.Generic.IList{PathAccessControlItem}, int?, System.IProgress{ChangeAccessControlListPartialResult}, System.Threading.CancellationToken)"/>
+    /// </summary>
+    public class ChangeAccessControlListResult
```

Comment:
Same comment about One SDK probably wanting to rename this, and adding a static method in ```models/DataLakeModelFactory.cs```

Reply:
@tg-msft Question about DataLakeModelFactory. Would exposing parametrized public constructor and making properties immutable be a viable alternative? I wonder why we've chosen Factory (harder to discover than constructor imo).

Reply:
That's exactly why we went with the factory.  😄  We've seen customers often get confused by trying to create a result type and then search for some a "fetch from server" method.  By only putting public .ctors on things users should construct, we're trying to guide them into success.  The model factory approach is our compromise to allow mocking for those who really want it without confusing everyone else (like a public factory .ctor might).

Reply:
Thanks for explanation. That makes sense (though I'm surprised by that fact).